src/rdd/rdd.rs

```diff
@@ -992,6 +994,68 @@ pub trait Rdd: RddBase + 'static {
         let min_fn = Fn!(|x: Self::Item, y: Self::Item| x.min(y));
         self.reduce(min_fn)
     }
+
+    fn top(&self, num: usize) -> Result<Vec<Self::Item>>
+    where
+        Self: Sized,
+        Self::Item: Data + Ord,
+    {
+        if num == 0 {
+            Ok(vec![])
+        } else {
+            let first_k_func = Fn!(move |partition: Box<dyn Iterator<Item = Self::Item>>|
+                -> Box<dyn Iterator<Item = BoundedMinPriorityQueue<Self::Item>>>  {
+                    let mut queue = BoundedMinPriorityQueue::new(num);
+                    partition.for_each(|item: Self::Item| queue.append(item));
+                    Box::new(std::iter::once(queue))
+            });
+
+            let queue = self
+                .map_partitions(first_k_func)
+                .reduce(Fn!(
+                    move |mut queue1: BoundedMinPriorityQueue<Self::Item>,
+                          queue2: BoundedMinPriorityQueue<Self::Item>|
+                          -> BoundedMinPriorityQueue<Self::Item> {
+                        queue1.merge(queue2);
+                        queue1
+                    }
+                ))?
+                .unwrap() as BoundedMinPriorityQueue<Self::Item>;
+
+            Ok(queue.into_vec_sorted())
+        }
+    }
+
+    fn take_ordered(&self, num: usize) -> Result<Vec<Self::Item>>
+    where
+        Self: Sized,
+        Self::Item: Data + Ord,
+    {
+        if num == 0 {
+            Ok(vec![])
+        } else {
+            let first_k_func = Fn!(move |partition: Box<dyn Iterator<Item = Self::Item>>|
+                -> Box<dyn Iterator<Item = BoundedMaxPriorityQueue<Self::Item>>>  {
+                    let mut queue = BoundedMaxPriorityQueue::new(num);
+                    partition.for_each(|item: Self::Item| queue.append(item));
+                    Box::new(std::iter::once(queue))
+            });
+
+            let queue = self
+                .map_partitions(first_k_func)
+                .reduce(Fn!(
+                    move |mut queue1: BoundedMaxPriorityQueue<Self::Item>,
+                          queue2: BoundedMaxPriorityQueue<Self::Item>|
+                          -> BoundedMaxPriorityQueue<Self::Item> {
+                        queue1.merge(queue2);
+                        queue1
+                    }
+                ))?
+                .unwrap() as BoundedMaxPriorityQueue<Self::Item>;
```

Comment:
we can proabbly avoid `unwrap` here altogether and use ? (it would like like a double question mark `?? `), if needed be you can add a new Error variant or use .map_err(|_| Error::{variant})?